### DISCUSSION TITLE
Component replaceWith(htmlString) not working as expected.

### DISCUSSION THREAD

**Discussion by AbdallahAbdedaiem** [GrapesJS/grapesjs#4874] [Q&A]
Hello,
I'm rendering some html out of grapesjs. then I want to use it to replace an existing component. I even tried adding hard coded html string but nothing happens and I have no errors. Maybe I' m doing something wrong.
Note: I logged the output of getEl method on gjsComponent to make sure it's a grapes js Component.
`gjsComponent.replacewith('<div>Bonjour les amis!'</div>');`

**Comment by artf** [comment]
`replaceWith` looks to work properly on my side (be careful with the uppercase `W`, as calling`replacewith` would result in `replacewith is not a function`). Are you able to create a reproducible demo?

**Reply by AbdallahAbdedaiem** [reply]
Hi @artf !
Thanks for the quick response!
I actually don't have such an issue. I even checked with:

`typeof gjsCompo.replaceWith // function`

I also tried styling the html of the gjsComponent to make sure I'm actually focusing the right area. so I added a bootstrap class and it worked as expected:

`gjsComponent.getEl().classList.add("bg-danger");// result: compo with red background in canvas
`

Previously, nothing occurred. Now, this is the error I get(grapes.min.js):
![Screenshot from 2023-01-30 17-14-28](https://user-images.githubusercontent.com/42803090/215533596-e4ad78b7-20c1-45ac-8228-5353012a760f.png)
![Screenshot from 2023-01-30 17-14-37](https://user-images.githubusercontent.com/42803090/215533602-e8ea1d38-f61a-41db-848e-b20584c0955a.png)

**Reply by artf** [reply]
Looks like you're trying to replace a component which is not yet added to the page, are you able to provide a reproducible demo?
